pkg/logs/decoder/decoder.go

```diff
@@ -97,15 +104,28 @@ func InitializeDecoder(source *config.LogSource, parser parsers.Parser) *Decoder
 // NewDecoderWithEndLineMatcher initialize a decoder with given endline strategy.
 func NewDecoderWithEndLineMatcher(source *config.LogSource, parser parsers.Parser, matcher EndLineMatcher, multiLinePattern *regexp.Regexp) *Decoder {
 	inputChan := make(chan *Input)
+	brokenLineChan := make(chan *DecodedInput)
+	parsedMessageChan := make(chan *Message)
 	outputChan := make(chan *Message)
 	lineLimit := defaultContentLenLimit
-	var lineHandler LineHandler
-	var lineParser LineParser
 	detectedPattern := &DetectedPattern{}
 
+	// construct the lineBreaker actor, wrapping the matcher
+	lineBreaker := NewLineBreaker(inputChan, brokenLineChan, matcher, lineLimit)
+
+	// construct the lineParser actor, wrapping the parser
+	var lineParser LineParser
+	if parser.SupportsPartialLine() {
+		lineParser = NewMultiLineParser(brokenLineChan, parsedMessageChan, config.AggregationTimeout(), parser, lineLimit)
```

Comment:
Do you think it would improve readability if this could be written as `NewMultiLineParser(lineBreaker.OutputChan(), ...)` ? (and similarly for other sub-actors/components)

Reply:
> I can see cases in general where that would be problematic, though -- if all actors create and supply their own output channels, then anything orchestrating those actors cannot "join" multiple streams into one.

Actors don't need to create the channels, just let them be queried. But if you don't like adding methods, maybe just `s/parsedMessageChan/lineParserOut/` to make it easier tell at a glance what's what.